src/gui/qgsmaptoolcapture.cpp

```diff
@@ -322,7 +332,10 @@ int QgsMapToolCapture::nextPoint( const QgsPoint& mapPoint, QgsPoint& layerPoint
   }
   try
   {
-    layerPoint = toLayerCoordinates( vlayer, mapPoint ); //transform snapped point back to layer crs
+    QgsPoint mapP( mapPoint.x(), mapPoint.y() );
+    layerPoint = QgsPointV2( toLayerCoordinates( vlayer, mapP ) ); //transform snapped point back to layer crs
+    if ( QgsWKBTypes::hasZ( QGis::fromOldWkbType( vlayer->wkbType() ) ) )
+      layerPoint.addZValue( 0.0 );
   }
```

Comment:
I think we should also add 

if ( QgsWKBTypes::hasM( QGis::fromOldWkbType( vlayer->wkbType() ) ) )
      layerPoint.addMValue( 0.0 );

here. Sure, it won't do anything until the WKB types mess is sorted, but when it is that will also fix adding geometries to layers with an m dimension